### ISSUE TITLE
Address finder not working in Safari

### ISSUE THREAD

**Issue by grantpezeshki** [nycehs/Heat_Hub#7]
Safari seems to only be dowloading part of ol.js - the openlayers mapping library. This means address search is no longer working in safari (mac and IOS default browser). 
Can try updating dependencies and also changing to a server (not cdn) copy of ol.js.
Also will need to upgrade location finder anyway; maybe should change to leaflet? review leaflet and mapbox pre-built components, if any available open source.

**Comment by mmontesanonyc** [comment]
Sounds like the rest of the besp team is starting to identify issues with the NYC Core Framework map/location finder, too, with a few failure points:

- Attempted ZIP entry
- Entering an intersection redirects the map but doesn't return a neighborhood
- Attempted dropping a pin
- Missed the second click required to get the data

Also, when we had worked on the AQE, I recall asking DoITT it it were possible to superimpose the NTA boundaries over the map (click to select?) to help further show the user what they're selecting. 

All of these would be nice functions of the location finder; not sure of the best way to address them.

**Comment by grantpezeshki** [comment]
Yes, I think we probably should send an issues list to the GIS team at DoITT that created it, but we also can pool a group of services ourselves. Some of these issues may not have a broad enough audience need for them to implement, and we can't force their timeline. I'm committed to the neighborhood reports at the moment but would be nice to release that with a better locator. 
We definitely should look at Planning software to see their implementations, as we might borrow.
Here are my implementation ideas: 
We need to find a more robust geocoding service that will return coordinates from zip - seems like geosupport (current geocoder) is not doing that, at least not consistently. Alternatively or in addition, we could talk to geosupport team to add in zip centroids as 'places' like battery park etc, so that works better. Planning has an NYC geocoder that may work, based on Pelias (OS geocoder), but we also could probably use an open street map geocoder etc.
We need to establish a javascript workflow that handles issues better. Right now intersections geocode, relocating the map window, but don't return UHF info, which we need for the rest of the page. In the case of no UHF we should use the Lat/Long to check UHF  - this is how we could also handle users dropping a point on the map. We could use turf.js, a javascript geoprocessing library, to handle this in the browser using a point-in-polygon operation.
The second click thing is a javascript problem that literally any real JS programmer could solve, but I need help with. Let's focus on the other issues first so that we pull the right stack together, and make our goal to release the new locator with the Neighborhood reports.
